tests/acceptance/features/bootstrap/WebUIGuestsContext.php

```diff
@@ -250,6 +250,108 @@ public function adminSetsValueOfBlockedDomainsSharingFromGuestsInputTo(string $b
 		$this->guestsPage->setBlockedDomainsFromSharingWithGuests($blockedDomains);
 	}
 
+	/**
+	 * @Given /^the checkbox to limit guest access to the whitelisted apps has been (enabled|disabled)$/
+	 *
+	 * @param string $state - whether enabled | disabled
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function theCheckboxToLimitGuestAccessToTheWhitelistedAppsHasBeen(string $state):void {
+		$this->guestsPage->setLimitGuestAccessToWhiteListedAppsCheckBox(strtolower($state) === "enabled");
+		$actualState = $this->guestsPage->getLimitGuestAccessToWhiteListedAppsCheckBox();
+		Assert::assertEquals(
+			$state,
+			$actualState,
+			"Expected checked state of checkbox for limiting guest access to whitelisted apps to be $state but found to be $actualState"
+		);
+	}
+
+	/**
+	 * @When /^the administrator (adds|removes) app "([^"]*)" (to|from) the whitelisted apps for guest access$/
+	 *
+	 * @param string $operation - adds | removes
+	 * @param string $appName - name of the app to be added to whitelist for guest access
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function theAdministratorAppToTheWhitelistedAppsForGuestAccess(string $operation, string $appName):void {
+		$operation = strtolower($operation);
+
+		$appsList = $this->guestsPage->getWhitelistedAppsForGuestAccess();
+		if ($operation === "adds") {
+			// the app is already present
+			if (\in_array($appName, $appsList)) {
+				return;
+			}
+			// Add to the array of whitelisted apps
+			$appsList[] = $appName;
+			$this->guestsPage->setWhitelistedAppsForGuestAccess(join(",", $appsList));
+			return;
+		}
+
+		// for app removal from guest access, filter the array
+		$appsList = \array_filter(
+			$appsList,
+			function (string $elem) use ($appName) {
+				return $elem !== $appName;
```

Comment:
use readable name for $elem
```suggestion
			function (string $elem) use ($appName) {
				return $elem !== $appName;
```